### ISSUE TITLE
Having a checkbox beside each item in the Wikidata schema window

### ISSUE THREAD

**Issue by antoine2711** [OpenRefine/OpenRefine#2438]
**Is your feature request related to a problem or area of OpenRefine? Please describe.**
A lot of the time, I will want to push to wikidata different parts of my dataset (i.e. row). Of course, you can delete and add back any declaration in the WD schema pane, but it would be VERY handy to be able to just « desactivate » a statement for one data transfer (or more). Having also the ability to desactivate a quantifier or a reference would be very good.

**Describe the solution you'd like**
For each declaration (label, desc, alias, statement, qualifier in statement, reference in statement, having a checkbox, whose value would be saved with the schema (export schema).

**Describe alternatives you've considered**
Doing a lot of import/export of schema.json, and some editing/merging sometimes.

**Additional context**
![image](https://user-images.githubusercontent.com/12217525/77052183-d3f6b280-69a2-11ea-8157-b961d906c745.png)

**Comment by wetneb** [comment]
I understand your use case, but I would think twice before adding that:
* This is effectively a redundant feature, so it is likely to be confusing for users who haven't been through the same thought process as you have;
* We will need space on statements to add more settings about them (add support for deleting existing statements on items, configuring the way added statements are matched with existing statements on the items) so this is going to become quite crowded.

**Comment by antoine2711** [comment]
> I understand your use case, but I would think twice before adding that:
> 
> * This is effectively a redundant feature, so it is likely to be confusing for users who haven't been through the same thought process as you have;

I'm not sure where the redundancy is here: it's really to make something more practical.

> * We will need space on statements to add more settings about them (add support for deleting existing statements on items, configuring the way added statements are matched with existing statements on the items) so this is going to become quite crowded.

Just having a checkbox (has shown in the screenshot) really take little place. And, in the logic of push data, this checkbox also opens the possibility of having a « calculated » value, something which would be very good with also being able to modify or delete statements.

**Comment by wetneb** [comment]
> I'm not sure where the redundancy is here: it's really to make something more practical.

It is redundant with the delete button for each statement.

> Just having a checkbox (has shown in the screenshot) really take little place.

The meaning of that checkbox might be obvious to you, because you added it yourself, but for others we need a label or at least a visual clue about what this does - and we need to make it clear what the difference is with the delete button.

> this checkbox also opens the possibility of having a « calculated » value

You can already do this: just blank the cells where you don't want a value to be added.

Perhaps you could implement this checkbox for yourself - I believe you could make an OpenRefine extension for it (which would patch the Wikidata extension in JS directly). This way people could add this checkbox on their own!

**Comment by antoine2711** [comment]
> Perhaps you could implement this checkbox for yourself - I believe you could make an OpenRefine extension for it (which would patch the Wikidata extension in JS directly). This way people could add this checkbox on their own!

That what what I thought first when I saw your resistance in the first reply. Got to love the flexibility.

But, let's go further into the details to see if this can benefit most. Let's say I have a table FilmsActeurs, which is actually all the distribution (actors of a movie, P161). In that table, I can push thu 3 axis: 1) obviously, [ « a film » « P161 distribution » « an actor » ], but also 2) [ « an actor » « P106 occupation » « Q10800557 acteur de cinéma » ] and, if you have the data, 3) [ « a fictional character » and many properties, like « instance of P31 », « gender P21 », « présent dans l'œuvre P1441 », 
Note that the extra data mentionned in 3) also enables [ « a film » « P161 distribution » « an actor » « QUANTIFIER P453 rôle » « a fictional character » ], which is just an extension of 1). 

OpenRefine is great in that all this can be done in one single operation on new data. That's because with new data, you have zero risk of creating duplicate data or contradictory data. When you push data in entities that already exist, for many reasons, it is wise to query the current data already on WD, check with you data, and push some of your data, or choose to keep the current data and not push anything.

Having a checkbox enable easy and fast « little pushing », where you activate or desactivate some of the WD schema elements. I'm sure if push little by little was easier, people would do it more.

> > I'm not sure where the redundancy is here: it's really to make something more practical.
> 
> It is redundant with the delete button for each statement.

Delete is a complete different thing in the way that it forces you to recreate the statement that can be quite complex.

> > Just having a checkbox (has shown in the screenshot) really take little place.
> 
> The meaning of that checkbox might be obvious to you, because you added it yourself, but for others we need a label or at least a visual clue about what this does - and we need to make it clear what the difference is with the delete button.

Of course, text could be added beside the checkbox (activate?), just like there is text for the trashcan (delete), and the greying of inactivated portion of the schema would make obvious interface clues. And I agree with you that there is a big need for a lot of features in that portion of the interface in the future. But I believe this activating is also a needed feature.

> > this checkbox also opens the possibility of having a « calculated » value
> 
> You can already do this: just blank the cells where you don't want a value to be added.

This answer is not very helpful. You might need that data, deleting it just because the schema feature works that way is not friendly. Beside, this calculation feature would be much more far reaching then just a « value != null », it could evaluate with data from the full cells[…] spectrum…

Let's see what other people see here. Thank you for your input, Antonin.

**Comment by wetneb** [comment]
> When you push data in entities that already exist, for many reasons, it is wise to query the current data already on WD, check with you data, and push some of your data, or choose to keep the current data and not push anything.

I totally agree we need to improve the workflows around this, but I am not sure such a checkbox would be the best way. I would prefer giving users more control about how statement matching is done, letting them choose between:
* do not add the statement if there is already a statement with that property (regardless of its value)
* do not add the statement if there is already a more precise value than the value we want to upload (for instance, do not add "instance of" "building" if there is already an "instance of" "castle")
* do not add the statement if there is already a statement with the same property and value (similar to what QuickStatement does)
* the current behaviour, taking into account qualifiers
and possibly other settings.

As a user I would find that more helpful than the ability to disable a statement.

**Comment by antoine2711** [comment]
> > When you push data in entities that already exist, for many reasons, it is wise to query the current data already on WD, check with you data, and push some of your data, or choose to keep the current data and not push anything.
> 
> I totally agree we need to improve the workflows around this, but I am not sure such a checkbox would be the best way. I would prefer giving users more control about how statement matching is done, letting them choose between:
> 
> * do not add the statement if there is already a statement with that property (regardless of its value)
> * do not add the statement if there is already a more precise value than the value we want to upload (for instance, do not add "instance of" "building" if there is already an "instance of" "castle")
> * do not add the statement if there is already a statement with the same property and value (similar to what QuickStatement does)
> * the current behaviour, taking into account qualifiers
> 
> and possibly other settings.
> 
> As a user I would find that more helpful than the ability to disable a statement.

Of course, I, as a user, would also prefer an enhanced create statement than this little checkbox functionality. I would also like much more to have the possibility to modify an existing statement, or even to delete an existing statement.

The added functionality to the creation of statement would make the schema pane contains much more information, and that would make a checkbox to enable/disable those schema information even more practical, as deleting that information would be more complicated to recreate.

Thanks again for your input. Regards.
